doc/interfaces/supercomputer_data/index.md

```diff
@@ -1,6 +1,114 @@
 ---
-title: "Supercomputer data interface"
+title: "Supercomputer data interface specification"
+date: "2024-01-02"
+author:
+  - name: "Sebastien Lenard"
+    orcid: "0000-0003-3358-7197"
+  - name: "Matt Fisher"
+    orcid: "0000-0003-3260-5445"
+citation: true
+listing:
+  type: "table"
+  contents:
+    - "*.md"
+  fields:
+    - "title"
+    - "description"
 ---
 
+The webapp data interface represents the **input** of the `snow_today_webapp_ingest`
+Python code in this repository.
+
 The supercomputer processes data and sends it, with SCP, to an NSIDC disk. From here,
 the webapp back-end picks it up to prepare it for visualization by the webapp.
+
+Allowing the supercomputer to be responsible for producing data and metadata, the webapp
+can focus on visualization and changes, for example adding new regions, won't depend on
+the webapp to be also updated.
+
+We need to to balance all of these concerns:
+
+- Webapp load time: startup load time, and load time when changing regions & variables
+- Webapp maintainability
+- Flexibility to change webapp behavior by pushing different data from the
+  supercomputer.
+- Predictability in how the webapp will respond to data changes
+
+
+## Revisions
+
+**Current version: 1.0.0beta1**
+
+{{< include .CHANGELOG.md >}}
+
+
+## Terms
+
+### RFC2119
+
+The terms **MUST**, **MUST NOT**, **SHOULD**, **SHOULD NOT**, **MAY**, and **OPTIONAL**
+are defined by [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+
+### Technical components
+
+* **webapp**: Because the webapp has a client and server component, avoid the confusing
+  term "front-end".
+* **webapp server** or **webapp back-end**: The thing that provides web-friendly data to
+  the front-end.
+* **webapp ingest**: The thing that turns supercomputer data into web-friendly data for
+  the webapp server. **This component _receives_ data over the interface specified by this
+  document.**
+* **supercomputer** or **supercomputer back-end**: Where the original data is produced.
+  **This component _sends_ data over the interface specified by this document.**
+
+:::{.callout-note}
+Only the latter two components are impacted by this specification.
+:::
+
+
+### User interface components
+
+* **Map**: A raster visualization of a region, including:
+    * **Shape** of the region
+    * A **gridded data variable**
+    * A **NoData mask** (optional)
+    * **Snow Water Equivalent point data** (optional)
+
+* **Plot**: A line chart visualization of data about a region, including:
+    * This water-year's measurements of a **plot data variable**
+    * Statistical measurements calculated from full climatology for the region:
```

Comment:
full climatology of the region (that is: the historical data of the same sensor/platform/variable ingested on the supercomputer side)

Reply:
7bd51f4